pkg/awsutils/awsutils.go

```diff
@@ -343,21 +350,73 @@ func (cache *EC2InstanceMetadataCache) initWithEC2Metadata() error {
 	}
 	log.Debugf("Found vpc-ipv4-cidr-block: %s ", cache.vpcIPv4CIDR)
 
-	// retrieve vpc-ipv4-cidr-blocks
+	// spawing go routine and retrieving vpc-ipv4-cidr-block
+	waitSGCh  := make(chan string)
+	waitVPCCh := make(chan string)
+	errorCh := make(chan error)
+	go wait.Forever(func() {
+		cache.initSGIDs(mac, waitSGCh, errorCh)
+		cache.initVPCIPv4CIDRs(mac, waitVPCCh, errorCh)
+	}, 30*time.Second)
+
+	for {
```

Comment:
This needs a short comment explaining the setup.

Reply:
Done!